Documentation/Communicational Patterns/2. Tooling/2.4. ObservedObject.md

```diff
@@ -0,0 +1,46 @@
+![document type: vision](https://themomax.github.io/resources/markdown-labels/document_type_vision.svg)
+
+# ObservedObject
```

Comment:
What would be the lifetime of an `ObservableObject`? Will it be created based on the first request to the endpoint and kept in memory until the connection is closed or could it also be a global single instance that can be shared between other `Component`s?

Reply:
The lifetime of an observable object is not managed by Apodini. I see two options for how to achieve this:
* `@ObservedObject` requires an auto-closure for its initial value, which can then be called to get the actual value each time the `Handler` is prepared to be evaluated (i.e. right before/after RequestInjectables are injected). The auto-closure could either return the same object each and every time or initialize a new one for each call.
* The transition from `Component` to `Handler` is made in a way that allows us to safely initialize a new instance of the according `Handler` each time we need one.

Reply:
I think this is a very important question. I would suggest to have the same approach as SwiftUI: `ObservedObjects` that are passed in from the outside could be shared across different components as sort of singletons. If an `ObservedObject` is created in the `Component` it is recreated every time a new connection is established but lives for the complete connection, e.g. in a client side stream the `ObservedObject` is staying around until the stream is closed by a response from the web service.

Reply:
I added a couple of sentences at the beginning of ObservedObject -> Lifetime to clarify this topic.